src/capturer/cosmos/__tests__/cosmos.test.ts

```diff
@@ -0,0 +1,352 @@
+/* eslint-disable no-underscore-dangle */
+import "@azure/cosmos";
+import {
+  ChangeFeedIteratorOptions,
+  ChangeFeedStartFrom,
+  Container,
+  CosmosClient,
+  StatusCodes,
+} from "@azure/cosmos";
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import { getChangeFeedIteratorOptions, processChangeFeed } from "../cosmos";
+import {
+  ContinuationTokenItem,
+  cosmosConnect,
+  getItemById,
+  upsertItem,
+} from "../utils";
+
+const error = new Error("Connection error");
+
+const mockCosmosClient: CosmosClient = {
+  database: jest.fn(),
+} as unknown as CosmosClient;
+
+const mockContinuationToken = { _tag: "Continuation", value: "token" };
+const mockBeginningIterator = { _tag: "Beginning" };
+
+jest.mock("@azure/cosmos", () => ({
+  CosmosClient: jest
+    .fn()
+    .mockImplementationOnce(() => mockCosmosClient)
+    .mockImplementationOnce(() => {
+      throw error;
+    }),
+  Container: jest.fn().mockImplementationOnce(() => mockContainer),
+  ChangeFeedStartFrom: {
+    Continuation: jest
+      .fn()
+      .mockImplementationOnce((_) => mockContinuationToken),
+    Beginning: jest.fn().mockImplementationOnce(() => mockBeginningIterator),
+  },
+  StatusCodes: {
+    NotModified: 304,
+  },
+}));
+
+describe("cosmosConnect", () => {
```

Comment:
Since this is a test related to a function placed in `utility.ts`, can we separate tests file in order to make more clear what we are testing?